dotCMS/src/integration-test/java/com/dotmarketing/portlets/htmlpages/business/render/HTMLPageAssetRenderedAPIImplIntegretionTest.java

```diff
@@ -0,0 +1,418 @@
+        when(request.getParameter(Host.HOST_VELOCITY_VAR_NAME)).thenReturn(this.host.getHostname());
+
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(host);
+
+        final HTMLPageAssetRenderedAPIImpl htmlPageAssetRenderedAPIImpl = new HTMLPageAssetRenderedAPIImpl();
+
+        final PageView pageRendered = htmlPageAssetRenderedAPIImpl.getPageRendered(
+                request, response, user, htmlPageAsset.getURI(), pageMode);
+
+        assertEquals(pageRendered.getPageInfo().getPage(), htmlPageAsset);
+        assertEquals(pageRendered.getSite(), host);
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPIImpl#getPageRendered(PageContext, HttpServletRequest, HttpServletResponse)
+     * When The host_id request's parameter is set but it does not exists, it should take hostsession attribute
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test
+    public void shouldUseDefaultHostWhenHostIdDoesNotExists() throws DotDataException, DotSecurityException {
+        init();
+        final PageMode pageMode = PageMode.ADMIN_MODE;
+        addPermission(role, host);
+        when(session.getAttribute( com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID )).thenReturn(host.getIdentifier());
+
+        when(request.getParameter("host_id")).thenReturn("not_exists_host_id");
+        when(request.getAttribute(com.liferay.portal.util.WebKeys.USER)).thenReturn(user);
+        when(request.getAttribute(WebKeys.CURRENT_HOST)).thenReturn(this.host);
+
+        final HTMLPageAssetRenderedAPIImpl htmlPageAssetRenderedAPIImpl = new HTMLPageAssetRenderedAPIImpl();
+
+        final PageView pageRendered = htmlPageAssetRenderedAPIImpl.getPageRendered(
+                request, response, user, htmlPageAsset.getURI(), pageMode);
+
+        assertEquals(pageRendered.getPageInfo().getPage(), htmlPageAsset);
+        assertEquals(pageRendered.getSite(), this.host);
+    }
+
+    /**
+     * Method to test: {@link HTMLPageAssetRenderedAPIImpl#getPageRendered(PageContext, HttpServletRequest, HttpServletResponse)
+     * When the {@link Host#HOST_VELOCITY_VAR_NAME} request's parameter is set, it should take this over session attribute
+     * Should return a right {@link PageView}
+     * @throws DotDataException
+     * @throws DotSecurityException
+     * @throws InterruptedException
+     */
+    @Test
+    public void shouldUseHostNameParameter()
+            throws DotDataException, DotSecurityException {
+        init();
+        final PageMode pageMode = PageMode.ADMIN_MODE;
+        addPermission(role, host);
+        final Host anotherHost = new SiteDataGen().nextPersisted();
+        when(session.getAttribute( com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID )).thenReturn(anotherHost.getIdentifier());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Unnecessary use of fully qualified name 'com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID' due to existing import 'com.dotmarketing.util.WebKeys'](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4718001)